lib/storage/file.js

```diff
@@ -796,6 +796,145 @@ File.prototype.getSignedUrl = function(options, callback) {
+ * @param {object[][]=} options.startsWith - Array of request parameters and 
+ *     their expected prefixes (e.g. [["$<field>", "<value>"]]). Values are 
+ *     translated into starts-with constraints in the conditions field
+ *     of the policy document (e.g. ["starts-with", "$<field>", "<value>"])
+ * @param {string=} options.acl - ACL for the object from possibly predefined 
+ *     ACLs
+ * @param {string=} options.successRedirect - The URL to which the user 
+ *     client is redirected if the upload is successfull
+ * @param {string=} options.successStatus - The status of the Google Storage 
+ *     response if the upload is successfull (must be string)
+ * @param {object=} options.contentLengthRange - Object providing 
+ *     minimum (options.contentLengthRange.min) and maximum 
+ *     (options.contentLengthRange.max) value for the request's 
+ *     content length
+ *
+ * @example
+ * file.getSignedPolicy({
+ *   equals: [["$Content-Type", "image/jpeg"]],
+ *   contentLengthRange: {min: 0, max: 1024},
+ *   expiration: Math.round(Date.now() / 1000) + (60 * 60 * 24 * 14) // 2 weeks.
+ * }, function(err, policy) {
+ *     // policy.string: the policy document, plain text
+ *     // policy.base64: the policy document, base64
+ *     // policy.signature: the policy signature, base64
+ * });
+ */
+File.prototype.getSignedPolicy = function(options, callback) {
+  if (options.expiration < Math.floor(Date.now() / 1000)) {
+    throw new Error('An expiration date cannot be in the past.');
+  }
+
+  var expirationString = new Date(options.expiration).toISOString();
+  var conditions = []; 
+
+  // Populate eq conditions
+  if (options.equals && options.equals instanceof Array) {
+    var equalsLength = options.equals.length;
+    for (var i = 0; i < equalsLength; i++) {
+      var equalCondition = options.equals[i];
+      if ((!(equalCondition instanceof Array)) || equalCondition.length !== 2) {
+        throw new Error('Equals condition must be an array of 2 elements.');
+      } else {
+        conditions.push(['eq', equalCondition[0], equalCondition[1]]);
+      }
+    }
+  }
+
+  // Populate starts-with conditions
+  if (options.startsWith && options.startsWith instanceof Array) {
+    var startsWithLength = options.startsWith.length;
+    for (var j = 0; j < startsWithLength; j++) {
+      var startCondition = options.startsWith[j];
+      if (!(startCondition instanceof Array) || startCondition.length !== 2) {
+        throw new Error('StartsWith condition must be an array of 2 elements.');
+      } else {
+        conditions.push(['starts-with', startCondition[0], startCondition[1]]);
+      }
+    }
+  }
+
+  // Eq condition for object name
+  conditions.push(['eq', '$key', this.name]);
+
+  // The bucket that will hold the object
+  conditions.push({'bucket': this.bucket.name});
+  
+  // The ACL for the file, if any provided
+  if (options.acl) {  
+    conditions.push({'acl': options.acl});
```

Comment:
No need to quote the key (for this and some other ones that follow). Also, please use a new line:

```js
conditions.push({
  acl: options.acl
});

conditions.push({
  success_action_redirect: options.successRedirect
});

// ...
```